MantidPlot/src/WindowFactory.h

```diff
@@ -0,0 +1,203 @@
+#ifndef WINDOWFACTORY_H
+#define WINDOWFACTORY_H
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidAPI/DllConfig.h"
+#include "MantidKernel/make_unique.h"
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/SingletonHolder.h"
+#include "Mantid/IProjectSerialisable.h"
+
+#include <cstring>
+#include <map>
+#include <vector>
+
+#ifndef Q_MOC_RUN
+#include <boost/shared_ptr.hpp>
+#endif
+
+//----------------------------------------------------------------------
+// Forward declarations
+//----------------------------------------------------------------------
+class ApplicationWindow;
+
+namespace Mantid {
+namespace API {
+
+/* Cross platform string case insensitive string comparator functor.
+ */
+struct CaseInsensitiveStringComparator {
+  bool operator()(const std::string &s1, const std::string &s2) const {
+#ifdef _MSC_VER
+    return _stricmp(s1.c_str(), s2.c_str()) < 0;
+#else
+    return strcasecmp(s1.c_str(), s2.c_str()) < 0;
+#endif
```

Comment:
Maybe it is worth checking with the person who wrote this in the first place, why they did it. It could well be that it is much more performant, although chances are that the boost version does the same thing just abstracted away.

I am not aware of a sensible place, but you could create one. If DynamicFactory uses the same thing it would have to be in `Kernel`, but it might be worth getting Martyn's opinion on this.